src/mod_roster.erl

```diff
@@ -883,11 +883,32 @@ remove_user(User, Server) ->
 remove_user(Acc, User, Server) ->
     LUser = jid:nodeprep(User),
     LServer = jid:nameprep(Server),
-    Acc1 = send_unsubscription_to_rosteritems(Acc, LUser, LServer),
-    R = mod_roster_backend:remove_user(LUser, LServer),
-    mongoose_lib:log_if_backend_error(R, ?MODULE, ?LINE, {User, Server}),
+    Acc1 = maybe_send_unsubscription_to_rosteritems(Acc, LUser, LServer),
+    Backends = mongoose_lib:find_behaviour_implementations(mod_roster),
+    lists:foreach(fun(Backend) ->
+            try
+                Backend:remove_user(LUser, LServer)
+            catch
+                Class:Reason ->
+                    StackTrace = erlang:get_stacktrace(),
+                    ?WARNING_MSG("event=cannot_delete_personal_data,backends = ~p,"
+                        "backend=~p,class=~p,reason=~p,stacktrace=~p",
+                        [Backends, Backend, Class, Reason, StackTrace])
+            end
+        end, Backends),
     Acc1.
 
+maybe_send_unsubscription_to_rosteritems(Acc, LUser, LServer) ->
```

Comment:
It should rather be named `try_send...` or `send...rosteritems_safe` because currently the name suggests that unsub. pushes are sent or not, depending on some condition, which is not the case. :)

Reply:
` try_send...` ++

Reply:
renamed to `try_send...`